API/Async.pm

```diff
@@ -377,6 +395,51 @@ sub getLatestCollectionTimestamp {
 	});
 }
 
+sub updateFavorite {
+	my ($self, $cb, $action, $type, $id) = @_;
+
+	$self->getToken(sub {
+		my ($token) = @_;
+
+		if (!$token) {
+			my $error = $1 || 'NO_ACCESS_TOKEN';
+			$error = 'NO_ACCESS_TOKEN' if $error !~ /429/;
+
+			$cb->({
+				name => string('Did not get a token' . $error),
+				type => 'text'
+			});
+		}
+		else {
+			my $countryCode = Plugins::TIDAL::API->getCountryCode($self->userId);
+
+=comment
+			# make sure we'll force an update check next time
+			my $updated = $self->updated;
+			$self->updated($updated . "$type:") unless $updated =~ /$type/;
+=cut
+			my $method = ($action =~ /add/) ? 'POST' : 'DELETE';
+			main::INFOLOG && $log->is_info && $log->info(uc($method) . " /users/", $self->userId, "/favorites/$type/$id?countryCode=$countryCode");
+
+			# no DELETE method in SimpleAsync
+			my $http = Slim::Networking::Async::HTTP->new;
+			my $request = HTTP::Request->new( $method => BURL . "/users/" . $self->userId . "/favorites/$type/$id?countryCode=$countryCode" );
+			$request->header( 'Authorization' => 'Bearer ' . $token );
+			$request->header( 'Content-Length' => 0 );
+			$http->send_request( {
+				request => $request,
+				onBody  => $cb,
+				onError => sub {
+					my ($http, $error) = @_;
+					$log->warn("Error: $error");
+					main::DEBUGLOG && $log->is_debug && $log->debug(Data::Dump::dump($http));
+					$cb->();
+				}
+			} );
+		}
```

Comment:
Absolutely untested. But something like the following might be more in line with `SimpleAsyncHTTP` with less custom code?

```perl
my $http = Slim::Networking::SimpleAsyncHTTP->new(
	$cb,
	sub {
		my ($http, $error) = @_;
		$log->warn("Error: $error");
		main::DEBUGLOG && $log->is_debug && $log->debug(Data::Dump::dump($http));
		$cb->();
	},{
	}
);

my $url = sprintf('%s/users/%s/favorites/%s/%s?countryCode=%s', BURL, $self->userId, $type, $id, $countryCode);
$http->_createHTTPRequest( 
	DELETE => $url,
	'Authorization' => 'Bearer ' . $token
);
```

Reply:
That said: I plan to add support for `DELETE` ASAP.